src/Console/Helper/TaskRunnerHelper.php

```diff
@@ -65,23 +65,30 @@ public function run(OutputInterface $output, TaskRunnerContext $context)
         }
 
         $taskResults = $this->taskRunner->run($context);
-
         $warnings = $taskResults->filterByResultCode(TaskResult::NONBLOCKING_FAILED);
+
         if ($taskResults->isFailed()) {
             $failed = $taskResults->filterByResultCode(TaskResult::FAILED);
-            return $this->returnErrorMessages($output, $failed->getAllMessages(), $warnings->getAllMessages());
+            $this->printErrorMessages($output, $failed->getAllMessages(), $warnings->getAllMessages());
+
+            $context->shouldHideCircumventionTip() || $this->printCircumventionTip($output);
+
+            return self::CODE_ERROR;
         }
 
-        if ($context->isSkipSuccessOutput()) {
-            $this->returnWarningMessages($output, $warnings->getAllMessages());
-            return self::CODE_SUCCESS;
+        if ($context->shouldSkipSuccessOutput()) {
```

Comment:
I added this guard to maintain the same logic, but reading it does not make sense to me. Surely the `shouldSkipSuccessOutput` flag should only control whether or not we `printSuccessMessage`. Why does it determine whether we print warnings?